### DISCUSSION TITLE
CSRF Timeout?

### DISCUSSION THREAD

**Discussion by michaeldiederich** [wheels-dev/wheels#1163] [Q&A]
Hi all,

when the users reuse our application after some minutes, there is a POST request with a CSRF Token.
After some minutes(?) of idle the request is not valid any more (not sure why right now) - could this be a problem with the CSRF Token? How long is the CSRF Token valid?

**Comment by michaeldiederich** [comment]
I guess I can skip this issue as it doesn't appear in my local dev Environment. Must be a production / server issue.

**Reply by bpamiri** [reply]
Check the session timeout settings between your local dev and the productions environment. By default the csrf authenticity token is stored in the session and will expire when the session expires. You can [configure](https://guides.cfwheels.org/cfwheels-guides/working-with-cfwheels/configuration-and-defaults#csrf-protection-settings) this token to be stored in a cookie instead if needed.

So you could simply be hitting a session timeout settings in production and/or you may need to look into sticky sessions on your load balancer if you have multiple servers in your production pool.
